sdk/core/core-http/src/httpHeaders.ts

```diff
@@ -28,10 +28,42 @@ export interface HttpHeader {
  */
 export type RawHttpHeaders = { [headerName: string]: string };
 
+export interface HttpHeadersLike {
+  set(headerName: string, headerValue: string | number): void;
+  get(headerName: string): string | undefined;
+  contains(headerName: string): boolean;
+  remove(headerName: string): boolean;
+  rawHeaders(): RawHttpHeaders;
+  headersArray(): HttpHeader[];
+  headerNames(): string[];
+  headerValues(): string[];
+  clone(): HttpHeadersLike;
+  toJson(): RawHttpHeaders;
+}
+
+export function isHttpHeadersLike(object?: object): object is HttpHeadersLike {
+  if (!object) {
+    return false;
+  }
+
+  const anyObj: any = object;
+  if (
```

Comment:
Do we need to check `headersArray`, `headerNames`, `headerValues`, and `toJson` as well?

Reply:
I waffled about this, since I wanted to be reasonably sure you were passing us the right thing but didn't want to make the checks super slow. @bterlson any gut feelings?

Reply:
My gut feeling was this is already checking so much that it's extremely unlikely to match some random object.

But I'm not super comfortable with that since those members are required on the interface.

I'm fine with not checking, though.

Reply:
I added the checks for now. From what I can tell nobody ever uses these helpers, so I'd like to clean them up in a later refactor